src/cdi_ndi/NDI_CP_Eloss.hh

```diff
@@ -0,0 +1,128 @@
+  NDI_CP_Eloss(const NDI_CP_Eloss &) = delete;
+
+  // >>> ACCESSORS
+
+  //! Get a stopping power.
+  double getEloss(const double temperature, const double density, const double v0) const;
+
+  //! Query to see if data is in tabular or functional form (true)
+  static constexpr bool is_data_in_tabular_form() { return true; }
+
+  //! Get the name of the associated data file
+  inline std::string getDataFilename() const { return std::string(); }
+
+  //! Get the material temperature grid.
+  sf_double getTemperatureGrid() const { return temperatures; }
+
+  //! Get the material density grid.
+  sf_double getDensityGrid() const { return densities; }
+
+  //! Get the projectile energy grid.
+  sf_double getEnergyGrid() const { return energies; }
+
+  //! Get the number of material temperature grid points.
+  size_t getNumTemperatures() const { return n_temperature; }
+
+  //! Get the number of material density grid points.
+  size_t getNumDensities() const { return n_density; }
+
+  //! Get the number of projectile energy grid points.
+  size_t getNumEnergies() const { return n_energy; }
+
+  //! Get the general eloss model type
+  rtt_cdi::CPModelType getModelType() const { return rtt_cdi::CPModelType::TABULAR_ETYPE; }
+
+private:
+// Only implemented if NDI is found
+#ifdef NDI_FOUND
+  void load_ndi();
+#endif
+
+private:
+  rtt_units::PhysicalConstexprs<rtt_units::CGS> pc; //!< Units
+
+  uint32_t n_energy;          //!< Number of gridpoints in projectile energy
+  uint32_t n_density;         //!< Number of gridpoints in target density
+  uint32_t n_temperature;     //!< Number of gridpoints in target temperature
+  double d_log_energy;        //!< Log spacing of projectile energy gridpoints
+  double d_log_density;       //!< Log spacing of target density gridpoints
+  double d_log_temperature;   //!< Log spacing of target temperature gridpoints
+  double min_log_energy;      //!< Log of minimum projectile energy
+  double min_log_density;     //!< Log of minimum target density
+  double min_log_temperature; //!< Log of minimum target temperature
+  double min_energy;          //!< Minimum target energy
+  double max_energy;          //!< Maximum target energy
+  double min_density;         //!< Minimum target density
+  double max_density;         //!< Maximum target density
+  double min_temperature;     //!< Minimum target temperature
+  double max_temperature;     //!< Maximum target temperature
+  sf_double energies;         //!< Vector of energy gridpoints
+  sf_double densities;        //!< Vector of density gridpoints
+  sf_double temperatures;     //!< Vector of temperature gridpoints
+  // Note that after unit conversions, *_energy is really *_speed
+
+  //! Storage for tabulated data
+  sf_double stopping_data_1d;
+
+  //! 3D indexing of 1D stopping power data
+  inline double get_stopping_data(int i_e, int i_d, int i_t) const {
+    return stopping_data_1d[i_t + n_temperature * (i_d + n_density * i_e)];
```

Comment:
@KineticTheory Because of the ongoing issues with `mdspan`, I just hardcoded this so we can get going.

Reply:
*nod* :)

Reply:
That sounds like a great plan.  It is very discouraging to follow the github page for `mdspan`.  It seems the current implementation is failing on many fronts and there isn't a lot of free energy to fix those issues.  We may need to wait until we can adopt a newer C++ standard.